app/controllers/api/posts_controller.rb

```diff
@@ -3,7 +3,27 @@ def create
     post = Post.new post_params
 
     if post.save
-      render json: :nothing, status: 200
+      render json: { id: post.id }, status: 200
+    else
+      render json: { errors: post.errors.full_messages }.to_json, status: 422
+    end
+  end
+
+  def update
+    post = Post.find(params[:id])
+
+    if post.published?
+      render json: 'Already published'.to_json, status: 422
```

Comment:
Yes, that's the reason. I don't want the API to have any business with published posts. Once it's published change it on the web interface. Mostly I want to prevent someone that for some reason gets access to an auth token to do much damage.